### PR TITLE
feat(onboarding): simplify and onboard to static and slice zone

### PR DESCRIPTION
**Pull request by BohdanOne**
**Resolves**: [DT-2246](https://linear.app/prismic/issue/DT-2246/adapt-static-and-slice-zone-ui)
**Resolves**: [DT-2247](https://linear.app/prismic/issue/DT-2247/display-about-static-fields-dialog-when-adding-a-slice-to-a-page)
### Description

- Adjustments to the Static Zone and Slice Zone UI: Name changes, buttons look. 
- Added `StaticFieldsInfoDialog` which is displayed on first click of button adding static field.

### Checklist

- [x] If my changes require tests, I added them.
- [X] Final modal content

### Preview

Screenshot 2024-08-13 at 10 17 33

Screenshot 2024-08-13 at 10 17 54

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

File: packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  );
+
+  return (
+    <StaticFieldsInfoContext.Provider
+      value={{ isInfoDialogSeen, setInfoDialogSeen }}
+    >
+      {children}
+    </StaticFieldsInfoContext.Provider>
+  );
+}
```
Comment (jomifepe) [inline packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: 💡 

I don't really see a reason to use context here, do we plan on opening the modal from somewhere else? Seems like having the `usePersistedState` be use directly in the `AddStaticFieldButtonComponent` would suffice.

Comment (BohdanOne) [reply packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: I had some issue with that, but will look into it again!

Comment (BohdanOne) [reply packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: removed!

File: packages/slice-machine/src/features/builder/AddFieldDropdown.tsx [hunk 2]
```diff
@@ -20,19 +21,25 @@ import {
...
 
 export function AddFieldDropdown(props: AddFieldDropdownProps) {
   const {
     disabled,
     onSelectField,
     fields,
-    triggerDataTestId = "add-field",
+    triggerDataTestId,
+    trigger,
+    ...implicitProps
```
Comment (levimykel) [inline packages/slice-machine/src/features/builder/AddFieldDropdown.tsx]: ❓ Why is `implicitProps` added here?

Comment (levimykel) [reply packages/slice-machine/src/features/builder/AddFieldDropdown.tsx]: I guess it's to pass `open` and `onOpenChange` to `DropdownMenu`. Why not explicitly pass those to make that more clear?

File: packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        onClose={dismissDialog}
+        trigger={!isInfoDialogSeen ? trigger : null}
+      />
+      <AddFieldDropdown
+        open={isAddFieldDropdownOpen}
+        onOpenChange={setAddFieldDropdownOpen}
+        disabled={disabled}
+        onSelectField={onSelectField}
+        fields={fields}
+        trigger={isInfoDialogSeen ? trigger : null}
```
Comment (xrutayisire) [inline packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: 💡 #best-practice

Here we have a best practice about not using null if undefined is possible. Same for others places where null is used in this PR.

```suggestion
        trigger={isInfoDialogSeen ? trigger : undefined}
```

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      size="small"
+    >
+      <DialogHeader title="Static zone" />
+      <DialogContent>
+        <ScrollArea>
+          <Box as="article" flexDirection="column" padding={16} gap={4}>
+            <Text sx={{ marginBottom: 4 }} color="purple9" variant="smallBold">
+              About
+            </Text>
+            <Text variant="h3">What is the Static zone</Text>
```
Comment (xrutayisire) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: 💡 #wording

Here there is "Static zone" but later on "static zone", we should have consitency, no?

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            />
+          </Box>
+        </ScrollArea>
+      </DialogContent>
+      <DialogActions>
+        <DialogActionButton
+          size="medium"
+          sx={{ flexGrow: 1 }}
+          onClick={dismissDialog}
+          data-testid="static-fields-info-dialog-cta"
```
Comment (xrutayisire) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: 💡 #idea

Can we not use just "Got it" text? We prefer to use text over test id when it can prevent us to update code. + it makes us test more like the user would, by first seeing the text and after clicking on it.

Edit: Much more because I see we query it like that already: 
```
await this.page
      .getByRole("button", { name: "Got it", exact: true })
      .click();
```

File: packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx [hunk 6]
```diff
@@ -55,6 +55,13 @@ export function UIDEditor() {
     setOpen(false);
   }
 
+  const uidIcon = useCallback(
```
Comment (xrutayisire) [inline packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:58]: ❓ #question

Why do we need that change?

File: packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx [hunk 7]
```diff
@@ -82,11 +90,8 @@ export function UIDEditor() {
               value={label}
               onValueChange={handleValueChange}
               error={error}
+              description="A label for the UID"
```
Comment (xrutayisire) [inline packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:93]: 🎉 #nice

So much better like that 🙂

File: packages/slice-machine/src/legacy/components/Navigation/index.tsx [hunk 8]
```diff
@@ -155,15 +144,6 @@ const Navigation: FC = () => {
           </SideNavListItem>
         )}
 
-        <ErrorBoundary>
-          <Suspense>
-            <VideoItem
```
Comment (xrutayisire) [inline packages/slice-machine/src/legacy/components/Navigation/index.tsx]: ❓ #question

What is this change related to? What is the product desire related to that?

File: packages/slice-machine/src/legacy/components/Navigation/index.test.tsx [hunk 9]
```diff
@@ -140,7 +140,7 @@ describe("Side Navigation", () => {
     expect(within(changesItem).getByText("1")).toBeVisible();
   });
 
-  test("Video Item with next", async (ctx) => {
+  test.skip("Video Item with next", async (ctx) => {
```
Comment (xrutayisire) [inline packages/slice-machine/src/legacy/components/Navigation/index.test.tsx]: 💡 #idea

Is it a permanent removal of the code related to that? Why not deleting related tests then?

File: packages/slice-machine/src/legacy/components/Navigation/index.tsx [hunk 10]
```diff
@@ -25,10 +24,6 @@ import { FolderIcon } from "@/icons/FolderIcon";
 import { LightningIcon } from "@/icons/Lightning";
 import { MasterSliceLibraryIcon } from "@/icons/MasterSliceLibraryIcon";
 import { SettingsIcon } from "@/icons/SettingsIcon";
-import VideoItem from "@/legacy/components/Navigation/VideoItem";
```
Comment (xrutayisire) [inline packages/slice-machine/src/legacy/components/Navigation/index.tsx]: ❓ #question

Should we remove related code?

File: packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+import {
+  AddFieldDropdown,
+  type AddFieldDropdownProps,
+} from "@/features/builder/AddFieldDropdown";
+import { StaticFieldsInfoDialog } from "@/features/builder/StaticFieldsInfoDialog";
+import { usePersistedState } from "@/hooks/usePersistedState";
+
+const LOCAL_STORAGE_KEY = "staticFieldsInfoDialogDismissed";
+
+export function AddStaticFieldButton(props: AddFieldDropdownProps) {
```
Comment (xrutayisire) [inline packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: 💡 #idea

I was a bit lost with the name. Here it's still a `AddFieldDropdown`, why not something like `AddStaticFieldDropdown`? We understand it's just a wrapper around `AddFieldDropdown` directly.

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx [hunk 12]
```diff
@@ -87,63 +88,76 @@ const Zone = ({
...
+                    // TODO(DT-1710): add the missing `flexShrink: 0` property to the Editor's Switch component.
+                    style={{ flexShrink: 0 }}
+                    data-testid="code-snippets-switch"
+                  />
+                </>
+              )
+            }
+            {
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+              zoneType === "slice" && fields.length > 0 ? (
```
Comment (xrutayisire) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx]: 💡 #idea

I would like if we can prevent using double ternary in our code, here it's old code, so I don't care much, but it's more a guideline for us.

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+        <ScrollArea>
+          <Box as="article" flexDirection="column" padding={16} gap={4}>
+            <Text sx={{ marginBottom: 4 }} color="purple9" variant="smallBold">
+              About
+            </Text>
+            <Text variant="h3">What is the Static zone</Text>
+            <Text color="grey11">
+              The static zone in Prismic page types contain fields that are
+              always present on the page. Use it for essential information like
+              a page's title, page design settings or metadata. Unlike slices,
```
Comment (levimykel) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: FYI, we use the oxford comma as part of our  [writing style guide](https://www.notion.so/prismic/Style-Guide-b1e1865ef5a54edeaf0d9ce6f4734d66?pvs=4#7face3f6c8a74615ad4582ce44559077).

"...a page's title, page design settings, or metadata."

File: playwright/tests/common/sideNav.spec.ts [hunk 14]
```diff
@@ -138,7 +138,7 @@ test.describe("Tutorial tooltip", () => {
     },
   });
 
-  test("I can close the tutorial video tooltip and it stays close", async ({
+  test.skip("I can close the tutorial video tooltip and it stays close", async ({
```
Comment (xrutayisire) [inline playwright/tests/common/sideNav.spec.ts]: 💡 #idea

Is it a permanent removal of the code related to that? Why not deleting related tests then?

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 4 above)
Comment (levimykel) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: Since we aren't referring to this as the Static zone anymore, I would change this title to match what we call it in the UI. For example, "What are static fields?"

File: playwright/pages/shared/TypeBuilderPage.ts [hunk 15]
```diff
@@ -226,6 +223,15 @@ export class TypeBuilderPage extends BuilderPage {
       .click();
   }
 
+  async dismissStaticFieldsInfoDialog() {
+    await this.page.getByTestId("add-field").click();
+    await this.page
+      .getByRole("button", { name: "Got it", exact: true })
+      .click();
+    await expect(this.addFieldDropdown.menu).toBeVisible();
+    await this.staticZoneAddFieldButton.click();
```
Comment (xrutayisire) [inline playwright/pages/shared/TypeBuilderPage.ts]: 💡 #idea

Finishing with button seems strange, much more how it's used later on. The function name is to dismiss static fields info dialog, but we are doing a side effect to click on the add field button at the end. I guess we could improve here and prevent side effect without assertion after.

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  return (
+    <Dialog
+      open={isOpen}
+      onOpenChange={handleOpenChange}
+      trigger={trigger}
+      size="small"
+    >
+      <DialogHeader title="Static zone" />
```
Comment (levimykel) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:29]: Since we aren't referring to this as the Static zone anymore, I would change this title to match what we call it in the UI, "Static fields".

File: packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+    />
+  );
+
+  return (
+    <>
+      <StaticFieldsInfoDialog
+        onClose={dismissDialog}
+        trigger={!isInfoDialogSeen ? trigger : null}
+      />
+      <AddFieldDropdown
```
Comment (levimykel) [inline packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: One of the requirements in the [linear ticket](https://linear.app/prismic/issue/DT-2247/display-about-static-fields-dialog-when-adding-a-slice-to-a-page) is to add a ? button near the CTA to re-open the info dialog. Was that requirement removed, or does this still need to be added?

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx [hunk 18]
```diff
@@ -87,63 +88,76 @@ const Zone = ({
...
+                <AddFieldDropdown {...addFieldDropdownProps} />
+              ) : zoneType === "customType" ? (
```
Comment (xrutayisire) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx]: ⚠️ #issue

For me there is an issue about when I'm seeing the new static zone info dialog, for custom types.

Even the message talk about page types, did product ask us to do both custom types AND page types?

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+        </ScrollArea>
+      </DialogContent>
+      <DialogActions>
+        <DialogActionButton
+          size="medium"
+          sx={{ flexGrow: 1 }}
+          onClick={dismissDialog}
+          data-testid="static-fields-info-dialog-cta"
+        >
+          Got it
```
Comment (xrutayisire) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:58]: ❓#question

Even if you never click on "Got it" you will not see it again after, is that wanted?

**Review by xrutayisire — CHANGES_REQUESTED** [review]
Nice work! Left some simple comments, didn't see anything big 🙂

File: packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx (hunk 3 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: I'm actually passing `null` to hide it. I could also create a prop to do this, but the idea is that `trigger` overrides the default one, but I also want to hide the default one and show nothing 😅

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 4 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: Yeah, changed 👍

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 5 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: Yeah it's not being used, but IMO test ids for E2E are better, that way we don't keep breaking the tests if we want to update a simple label, only the full flow matters.

File: packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx (hunk 6 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:58]: To update the icon size, unfortunately the `renderStartIcon` accepts a component definition, which IMO is wrong, it should accept a `ReactNode`, but for that reason, I define one and memoize it.

File: packages/slice-machine/src/legacy/components/Navigation/index.tsx (hunk 8 above)
Comment (jomifepe) [reply packages/slice-machine/src/legacy/components/Navigation/index.tsx]: This was a request from Côme, but I extracted a separate PR for it: https://github.com/prismicio/slice-machine/pull/1428

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 5 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: I don't agree to that, I prefer to have text when text is part of your product. It's our case on our side, with exact, it allows us to prevent someone to make a change of text without wanting to. I remember one time when I added a "s" because I wanted to do CTRL + s to save and e2e tests broke to warn me about it. 
When your text is part of your product, I would always favor text.

File: packages/slice-machine/src/legacy/components/Navigation/index.test.tsx (hunk 9 above)
Comment (jomifepe) [reply packages/slice-machine/src/legacy/components/Navigation/index.test.tsx]: Same as https://github.com/prismicio/slice-machine/pull/1426#discussion_r1715061859, but yeah I agree that they can be removed ✅

File: packages/slice-machine/src/legacy/components/Navigation/index.tsx (hunk 10 above)
Comment (jomifepe) [reply packages/slice-machine/src/legacy/components/Navigation/index.tsx]: Can do that, I didn't do it right away because I see that that it was quite a few lines of code, and we could want to bring it back in the future 🤷‍♂️

Will do it here: https://github.com/prismicio/slice-machine/pull/1428

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 19 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:58]: Not sure I understood, if you dismiss the modal it won't be opened again on add static field.

File: packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx (hunk 3 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: Oh ok, got it! 🙌

File: packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx (hunk 6 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:58]: Ahhh ok, I don't like that we changed the default, like icon size small if for button size small normally 😬
But btw, I'm not a fan of use useCallback just for the sake of using it without a reason.

Do we have a performance problem? React don't encourage using it if not for performance optimization.
See: https://react.dev/reference/react/useCallback

Screenshot 2024-08-13 at 13 35 11

`useCallback` cost perf to do the memoize, and sometimes it's costing more than what you saved.

Also: [Should you add useCallback everywhere? ](https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere)

I let you read this part and let you tell me what you think for this use case, if you think we should still wrap it or not necessary. It something I always like to learn more and more about when its good / bad to do.

Screenshot 2024-08-13 at 13 37 08

File: packages/slice-machine/src/legacy/components/Navigation/index.test.tsx (hunk 9 above)
Comment (xrutayisire) [reply packages/slice-machine/src/legacy/components/Navigation/index.test.tsx]: Prevent redoing the same comment there but yeah If product wants us to remove the menu item totally, let's remove ALL code related to that, components, tests, everything

File: packages/slice-machine/src/legacy/components/Navigation/index.tsx (hunk 10 above)
Comment (xrutayisire) [reply packages/slice-machine/src/legacy/components/Navigation/index.tsx]: Same:
> Prevent redoing the same comment there but yeah If product wants us to remove the menu item totally, let's remove ALL code related to that, components, tests, everything


I guess it depends on what Côme said, if they want to remove it totally and don't know about a usage again, let's remove it totally. We can find it after anyway with git history

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 19 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:58]: What I mean is when you don't click on "Got it" but close the modal with another way like the close icon, it will still be taken as "Got it". Just wanted to be sure it's ok for product.

File: packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx (hunk 11 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: I guess that was the name chosen because the only visible element without interaction is a button

Comment (jomifepe) [reply packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: But I'm okay with that, renamed

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx (hunk 12 above)
Comment (jomifepe) [reply packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx]: Done

File: packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx (hunk 6 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:58]: I'm aware of the premature optimization issue with `useMemo` and `useCallback`, but I believe defining components at render time is a worse pitfall than that.

Comment (xrutayisire) [reply packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:58]: Good then, wanted to be sure you considered that 🙂

Comment (xrutayisire) [reply packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:58]: Btw it will still be defined at render time, that didn't avoid that, just the number of time you're doing it

Comment (jomifepe) [reply packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:58]: Yeah, I would prefer to fix this issue in the editor side, accepting a definition is not justifiable for an icon 😅

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 19 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:58]: Ah, will check that

File: packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx (hunk 6 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx:58]: It as discussed a lot in the past, you get the Icon in param, that's why it's like this. But I don't remember everything. Agree could be good to refresh on the reasons haha. Anyway good for here

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx (hunk 18 above)
Comment (jomifepe) [reply packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx]: Confirmed this with product, made the change to only display the modal for page types.

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+      size="small"
+    >
+      <DialogHeader title="Static zone" />
+      <DialogContent>
+        <ScrollArea>
+          <Box as="article" flexDirection="column" padding={16} gap={4}>
+            <Text sx={{ marginBottom: 4 }} color="purple9" variant="smallBold">
+              About
+            </Text>
+            <Text variant="h3">What is the static zone?</Text>
```
Comment (levimykel) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:36]: Since we aren't referring to this as the Static zone anymore, I would change this to match what we call it in the UI, "What are static fields?".

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+            <Text color="grey11">
+              The static zone in Prismic page types contain fields that are
```
Comment (levimykel) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:38]: Since we aren't referring to this as the Static zone anymore, I would change this to match what we call it in the UI, "The static fields in Prismic...".

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+              always present on the page. Use it for essential information like
+              a page's title, page design settings or metadata. Unlike slices,
+              which can be added or removed, static zone fields remain fixed.
```
Comment (levimykel) [inline packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:41]: Since we aren't referring to this as the Static zone anymore, I would change this to match what we call it in the UI, "..., static fields remain fixed."

File: packages/slice-machine/src/features/builder/AddStaticFieldDropdown.tsx [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+
+  const trigger = (
+    <Button
+      {...commonTriggerProps}
+      {...(triggerDataTestId != null && { "data-testid": triggerDataTestId })}
+    />
+  );
+
+  return (
+    <>
```
Comment (levimykel) [inline packages/slice-machine/src/features/builder/AddStaticFieldDropdown.tsx]: ⚠️ There's no need to wrap this in fragments, I think.

**Review by levimykel — COMMENTED** [review]
Looks good, I added some questions and comments.

File: packages/slice-machine/src/features/builder/AddFieldDropdown.tsx (hunk 2 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/AddFieldDropdown.tsx]: Unsure of the reason, but I'm okay with that change 😄

Comment (jomifepe) [reply packages/slice-machine/src/features/builder/AddFieldDropdown.tsx]: Done

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 13 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: Ah, makes sense, changed

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 4 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: Make sense to me, will check with Côme

Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx]: Missed the final correction from Côme, and we're not changing anything to Static fields after all, so these labels make sense, will change the only `Static fields` one back to `Static zone`.

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 16 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:29]: https://github.com/prismicio/slice-machine/pull/1426#discussion_r1715315531

File: packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx (hunk 17 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx]: Still checking this with Benjamin, there's no design for it yet.

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 21 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:38]: This clears this up: https://github.com/prismicio/slice-machine/pull/1426#discussion_r1715315531

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 22 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:41]: This clears this up: https://github.com/prismicio/slice-machine/pull/1426#discussion_r1715315531

File: packages/slice-machine/src/features/builder/AddStaticFieldDropdown.tsx (hunk 23 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/AddStaticFieldDropdown.tsx]: Yeah fixed 😄

File: packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx (hunk 19 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/builder/StaticFieldsInfoDialog.tsx:58]: Checked with Benj & Côme and now closing the dialog doesn't prevent it from showing up the next time

File: playwright/pages/shared/TypeBuilderPage.ts (hunk 15 above)
Comment (jomifepe) [reply playwright/pages/shared/TypeBuilderPage.ts]: Refactored, tell me what you think

Comment (xrutayisire) [reply playwright/pages/shared/TypeBuilderPage.ts]: This `getStaticZoneInfoDialogConfirmCta` can be a simple static field as you don't pass param, but looks good otherwise

Comment (xrutayisire) [reply playwright/pages/shared/TypeBuilderPage.ts]: I let you do the change and you can resolve the comment yourself

Comment (jomifepe) [reply playwright/pages/shared/TypeBuilderPage.ts]: Won't that fail? Since it will look for it right away

**Review by levimykel — APPROVED** [review]
Nice work 😄

File: playwright/pages/shared/TypeBuilderPage.ts (hunk 15 above)
Comment (xrutayisire) [reply playwright/pages/shared/TypeBuilderPage.ts]: No it doesn't, that's the beauty of it 🤩

Comment (xrutayisire) [reply playwright/pages/shared/TypeBuilderPage.ts]: Dynamic field like you did is only when you want, for example an item with a name that you don't have before the test. But everything in our classes will only be executed when you call them in a test.

Comment (jomifepe) [reply playwright/pages/shared/TypeBuilderPage.ts]: Awesome then, refactored, thanks for the suggestion 🚀

**Review by xrutayisire — APPROVED** [review]
LGTM! Nice work to both you!

🎉
